### PR TITLE
Tests and fixes for rt::Bytes

### PR DESCRIPTION
**Pull request by bbannier**

### CODE REVIEW

File: hilti/include/rt/types/bytes.h [hunk 1]
```diff
@@ -297,7 +337,7 @@ class Bytes : protected std::string, public hilti::rt::detail::iterator::Control
     Bytes join(const Vector<T>& parts) const {
         Bytes rval;
 
-        for ( auto i = 0; i < parts.size(); ++i ) {
+        for ( size_t i = 0; i < parts.size(); ++i ) {
```
Comment (rsmmr) [inline hilti/include/rt/types/bytes.h:340]: Just curious: Why this change?

File: tests/hilti/types/bytes/iterator.hlt [hunk 2]
```diff
@@ -18,15 +18,6 @@ assert *i++ == '1';
 assert *i++ == '2';
 assert *++i == '4';
 
-assert i > j;
```
Comment (rsmmr) [inline tests/hilti/types/bytes/iterator.hlt:21]: Did we lose the comparison operators? They would be good to keep, so that one can build custom loops.

File: hilti/include/rt/types/bytes.h (hunk 1 above)
Comment (bbannier) [reply hilti/include/rt/types/bytes.h:371]: As written `i` is signed while `parts.size()` is unsigned. This triggers a diagnostic e.g., with clang that we previously didn't see as we only exercised this code in JIT; now this causes an compilation error from the unit test.

File: tests/hilti/types/bytes/iterator.hlt (hunk 2 above)
Comment (bbannier) [reply tests/hilti/types/bytes/iterator.hlt:21]: This is the only place we ever exercised this kind of operator for operators, and we e.g., do not support it for `rt::Vector` iterators either. General STL-type iterators are typically only compared for inequality (e.g., `it != container.end()`), but if we need iterators to be ordered as well I can add the operators back. In that case lets come up with some code which actually exercises that property.

Comment (bbannier) [reply tests/hilti/types/bytes/iterator.hlt]: I added the ordering of iterators into `rt::Bytes` back (along with the previously removed test case), and also opened #363 for adding the same property to iterators into `rt::Vector`. As `rt::List` is not index-based I didn't add this property to its iterators, but if it turns out to be important we can change its iterators as well to hold the needed internal state.
